README.md

```diff
@@ -70,6 +68,58 @@ libsecp256k1 is built using autotools:
 
 To compile optional modules (such as Schnorr signatures), you need to run `./configure` with additional flags (such as `--enable-module-schnorrsig`). Run `./configure --help` to see the full list of available flags.
 
+Building with CMake
+-------------------
+
+To maintain a pristine source tree, CMake encourages to perform an out-of-source build by using a separate dedicated build tree.
+
+### Building on POSIX systems
+
+    $ rm -rf build && mkdir build && cd build
+    $ cmake ..
+    $ make
+    $ sudo make install  # optional
+
+To adjust the build system configuration, the following options can be provided to `cmake` call:
+
+| Option | Default value | Description |
+|--------|:-------------:|-------------|
+| BUILD_BENCHMARK | ON | compile benchmark |
+| BUILD_TESTS | ON | compile tests |
+| BUILD_EXHAUSTIVE_TESTS | ON | compile exhaustive tests |
+| BUILD_EXAMPLES | OFF | compile examples |
+| ENABLE_MODULE_ECDH | OFF | enable ECDH module |
+| ENABLE_MODULE_RECOVERY | OFF | enable ECDSA pubkey recovery module |
+| ENABLE_MODULE_EXTRAKEYS | OFF |enable extrakeys module |
+| ENABLE_MODULE_SCHNORRSIG | OFF | enable schnorrsig module |
+
+For example:
+
+    $ cmake -DBUILD_BENCHMARK=OFF -DENABLE_MODULE_ECDH=ON ..
+
+### Cross compiling
+
+To alleviate cross compiling, preconfigured toolchain files are available in the `toolchain` directory.
+For example, to cross compile for Windows:
+
+    $ cmake -DCMAKE_TOOLCHAIN_FILE=../toolchain/x86_64-w64-mingw32.cmake ..
+
+To compile for Android with NDK (not using a toolchain file, and assuming the `ANDROID_HOME` environment variable has been set):
```

Comment:
I think the correct variable is `ANDROID_NDK_ROOT`, see https://github.com/openssl/openssl/issues/11205 and the docs https://cmake.org/cmake/help/latest/manual/cmake-toolchains.7.html#cross-compiling-for-android . Maybe it makes sense to have a link to the docs, or only have that link. Not sure what others think.

Reply:
Thanks! [Updated](https://github.com/bitcoin-core/secp256k1/pull/1113#issuecomment-1170397983).